### PR TITLE
draft: New topic - mimblewimble, all the bits

### PR DESCRIPTION
**Pull request by CjS77**
This post describes the key elements of _vanilla_ Mimblewimble, what purpose they perform, and the bad things that could happen if they're not there.

It also describes a few pieces that were introduced to the protocol as a result of to omissions and oversights in
the MW whitepaper.

### CODE REVIEW

File: _posts/course5-mimblewimble/2021-05-03-mimblewimble-bits.md [hunk 1]
```diff
@@ -18,11 +18,379 @@ The Mimblewimble white paper provided the foundations of the protocol. During th
...
+| Data               | Malleable? | Reason                                                                                     |
+|--------------------|------------|--------------------------------------------------------------------------------------------|
+| Range proof        | No         | Constructing a range proof requires knowledge of the blinding factor and commitment value. |
+| UTXO value         | No         | The consensus balance will detect a changed UTXO value.                                    |
+| UTXO spending      | No         | The kernel signature will be invalid. The consensus balance will fail.                     |
+| Transaction offset | No         | The consensus balance will fail.                                                           |
+| UTXO features      | Yes        | If an output is cut-through, there is no record of the features.                           |
+| Header data        | Yes        | See comment below                                                                          |
+
+Note: Tari resolves the UTXO feature set malleability by disabling cut-through. In general though, UTXO features are
```
Comment (SWvheerden) [inline _posts/course5-mimblewimble/2021-05-03-mimblewimble-bits.md:169]: Technically not entirely true, 
Tari resolves the UTXO feature set malleability by adding a signature, called the `metadata signature` on the utxo signed by the blinding factor `k_s`

This signature is also signed by `k_o`, and `k_o` is protected by the same method that prevents cut-through.

**Review by SWvheerden — COMMENTED** [review]
Looks good

**Comment by SWvheerden** [comment]
The only thing I think of what can do is perhaps adding a piece of what we changed on Tari,
aka scripts and the script offset. 

We never talk about scripts, but we do talk about preventing cut-through but we don't link to that or explain it.

File: _posts/course5-mimblewimble/2021-05-03-mimblewimble-bits.md (hunk 1 above)
Comment (CjS77) [reply _posts/course5-mimblewimble/2021-05-03-mimblewimble-bits.md:169]: Yeah, I'll add that bit of nuance.

**Comment by CjS77** [comment]
> 

It might be better to have a separate TLU topic on that.

We've got the content of the TariScript paper that can form part of that, or even be its own article

**Comment by SWvheerden** [comment]
mm yeah, perhaps a link to the RFC or the paper is more than sufficient.

File: _posts/course5-mimblewimble/2021-05-03-mimblewimble-bits.md (hunk 1 above)
Comment (CjS77) [reply _posts/course5-mimblewimble/2021-05-03-mimblewimble-bits.md:169]: Of course, if we didn't disable cut-through, those signatures could also be discarded, ceteris parabis, so arguably stopping cut-through is the key intervention here.

**Pull request merged**